geo/src/algorithm/linestring_segment.rs

```diff
@@ -0,0 +1,186 @@
+use crate::line_interpolate_point::LineInterpolatePoint;
+use crate::{Coord, Densify, EuclideanLength, LineString, LinesIter, MultiLineString};
+
+/// Segments a LineString into `n` equal length LineStrings as a MultiLineString.
+/// `None` will be returned when `n` is equal to 0 or when a point
+/// cannot be interpolated on a `Line` segment.
+pub trait LineStringSegmentize {
```

Comment:
We like to have a concise example in the form of a doctest for traits, under the `Examples` header. See e.g. `Densify`.